### PR TITLE
GH: Added a guide for setting up LAN deathmatch.

### PR DESCRIPTION
**Pull request by Zungrysoft**
### Description of Changes
---
Added a guide for setting up LAN deathmatch. Contains useful console commands and information about maps. The guide is in `docs/deathmatch-setup-guide.txt`.

### Checklist
---

- [x] I have read the LibreQuake contribution guidelines
- [x] I have thoroughly tested my changes to the best of my ability
- [x] I confirm I have not contributed anything that would impact LibreQuake's licensing and usage
- [ ] This Pull Request fixes a **critical** issue that should be reviewed and merged as soon as possible

### CODE REVIEW

File: docs/deathmatch-setup-guide.txt [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+map lqdm6	Ghost Quarter: A castle map where new weapons and armor types unlock every 40 seconds.
+map lqdm7	Boomstick Basement: A claustrophobic map with a labyrinthine structure.
+map lqdm8	Alichar Sector: A very tall map with a long elevator to reach the Red Armor.
+map lqdm9	Outlands: A temple map designed for 4v4 team deathmatch. Use the grenade launcher to unlock secret doorways!
+map lqdm10	Lucifer's Gambit: An evil temple with Lucifer's Protection as the main powerup.
+map lqdm11	The Wicked Night: A medium-sized map built around a central nexus.
+map lqdm12	Javelin: Contains a respawning key you can use to permanently unlock new rooms and powerups!
+map lqdm13	Death Provides No Return: A void map with jump pads and toggling windows.
+
+Additionally, all of the singleplayer levels are compatible with deathmatch. These maps are great if you have a player count of five or above since the gameplay spaces are much larger. If you want to switch to the third map in Episode 1, you would use the command "map e1m3".
```
Comment (MotoLegacy) [inline docs/deathmatch-setup-guide.txt:59]: I believe the correct command would be `changelevel` to keep the server running. Can you confirm?

Comment (Zungrysoft) [reply docs/deathmatch-setup-guide.txt:59]: It's different between different engines. In FTEQW both commands keep the server running.

Comment (lavenderdotpet) [reply docs/deathmatch-setup-guide.txt:59]: just change it to changelevel for engine constancy and then we should be good

File: docs/deathmatch-setup-guide.txt [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+Time Limit:		10 Minutes
+Frag Limit:		10 Frags
+Map:			lqdm1
+Leave the other settings alone.
+
+Then press "Start Game".
+
+Once you're in the server, you'll need to share your local IP Address with the other players. Open your operating system's terminal. (PowerShell on Windows, Terminal on MacOS, etc.) Type "ipconfig". This will bring up a list of IP Addresses for your computer. Find the one called "IPv4 Address". It will usually look something like "192.168.0.22".
+
+To connect to the server, each other player will need to open the console and type "connect", followed by your local IP Address. For example, "connect 192.168.0.22". (Open the console either by pressing Esc on the main menu or by pressing ~ while in game.)
```
[the remark this answers is not in the document]
Comment (Zungrysoft) [reply docs/deathmatch-setup-guide.txt:18]: Ah, I didn't realize that you could find LAN games in the browser. I'm updating the guide.

File: docs/deathmatch-setup-guide.txt [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+================
+Deathmatch Modes
+================
+To change the deathmatch mode, use the "deathmatch" command. LibreQuake has five different deathmatch modes which change the way the game plays. Deathmatch 3 is the recommended default mode.
+
+deathmatch 1	Items and weapons respawn. Mostly used for 4v4 team deathmatch.
+deathmatch 2	Items never respawn; weapons respawn instantly. This mode is rarely used.
+deathmatch 3	Items respawn; weapons respawn instantly. Great for free-for-all and team deathmatch.
+deathmatch 4	Spawn with all weapons, full health and armor, and infinite ammo. Collect backpacks for bonus health.
+deathmatch 5	Spawn with all weapons, full health and armor, and some ammo.
```
[the remark this answers is not in the document]
Comment (Zungrysoft) [reply docs/deathmatch-setup-guide.txt:71]: Yes. those modes are all available in QW.

**Comment by Zungrysoft** [comment]
@pnahratow 

> I'm not sure who the target audience of this guide is but it jumps from super basic. "Be connected to same network, go to Online -> Start Game" to console commands and server info keys.

I've reworked the guide a little bit to hopefully keep all of the advanced stuff separate from the basics. Thanks for your input!

> I for example wanted to know about how to make a map rotation.

I didn't thing to add this because I figured it wasn't very useful for LAN deathmatch. I'll add a section at the bottom of the infokeys section though.

**Review by MotoLegacy — APPROVED** [review]
Thanks for your work on this, looks good to me. In the future don't worry about merging main into your branch if your branch isn't conflicting since it adds commits :)

**Comment by lavenderdotpet** [comment]
@Zungrysoft you ready for this to be merged?

**Pull request merged**